cuqi/utilities/_utilities.py

```diff
@@ -229,3 +231,88 @@ def approx_gradient(func, x, epsilon= 0.000001):
         eps_vec[i] = 0.0
         
     return FD_gradient
+
+
+# Function for plotting 1D density functions
+def plot_1D_density(density:Density,
```

Comment:
I suggest adding a parameter to allow plot in log scale

Reply:
Agree to allow log as well

Reply:
good point! Added